### ISSUE TITLE
Empty string coercion in ID scalar type 

### ISSUE THREAD

**Issue by OlegIlyenko** [facebook/graphql#404]
I was reading this section of the spec and could not find an explicit info about handling of empty string values:

http://facebook.github.io/graphql/October2016/#sec-ID

Given that ID type supposed to represent some kind of identifier and input coercion rules allow to use a number literal, I wonder whether it makes sense to explicitly define a rule for an empty string coercion (in particular, do not allow it). This particular aspect can also be viewed as a part of the semantic analysis, so I guess it might also be represented as a query validation rule (though value can be provided as a variable, so it can't really be statically validated).

So my question is (based on my understanding, spec leaves it open for interpretation): **do you think it makes sense to add a rule that disallows empty string values for an ID scalar type?**

Some context: this issue was recently [raised in sangria](https://github.com/sangria-graphql/sangria/pull/332) and I would like to get a deeper understading of this topic.

**Comment by leebyron** [comment]
Maybe so! I'd like to keep the definition as broad as possible in spec, while implementing libraries are more than welcome to be more strict if it's beneficial. I'd like to avoid specifying any particular ID format. Perhaps we should specifically disallow the 0-length string, but if "empty" means a string that contains whitespace like the example you linked then I'm not so sure - some ID format might want to use whitespace, and I'm not sure it's valuable to declare that "not valid GraphQL" via spec restriction. What do you think?

**Comment by BjRo** [comment]
We implemented this now by adding a custom schema validation rule after some discussion with @OlegIlyenko . I'm not sold on the usefulness of empty or all whitespace strings as IDs, but from the spec/framework perspective I understand that you want to be as open as possible. 

I think the solution we arrived now is a good tradeoff. It's fairly easy to add while still allows the id definition to be very loose. From my side this can be closed!
